### PR TITLE
Add problem 11

### PR DESCRIPTION
**Pull request by Achilleshiel**
Generator: DirectionSelector generate coordinates for vertical, horizontal, and diagonal up and down positions in a grid.
Consumer: Maxer
Filter: SelectDigit, reads a value from a grid depending on the coordinates given.
Filter: Producter, gives the product of a series of values

Changed the digit_reader to be able to read more than one digit

### CODE REVIEW

File: filter/producter.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package filter
+
+// Produter multiplies n numbers.
+type Produter struct {
```
Comment (silkeh) [inline filter/producter.go]: This is missing a `c`: `Producter`

Comment (silkeh) [reply filter/producter.go]: Also: what products are supplied on the output: the product of the last `n`, or a set of `n`?

File: filter/select_digit.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
+package filter
+
+import "strconv"
+
+// SelectDigit generates a digit from a given grid of values
```
Comment (silkeh) [inline filter/select_digit.go]: A filter shouldn't generate

File: generator/direction_selector.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+			//Vertical
+			if x <= verx && y <= very {
+				dg.VerticalGenerator(out, x, y)
+			}
+		}
+	}
+}
+
+// DiagonalUpGenerator outputs a series of x, y positions
+func (dg *DirectionGenerator) DiagonalUpGenerator(out chan<- int, i, j int) {
```
Comment (silkeh) [inline generator/direction_selector.go]: All methods on the `*DirectionGenerator` should be private.

**Review by silkeh — CHANGES_REQUESTED** [review]
Please rebase on the latest master for the CI!

**Comment by Achilleshiel** [comment]
That should fix it.

**Review by silkeh — COMMENTED** [review]
LGTM!

**Review by silkeh — APPROVED** [review]

**Pull request merged**